Source/bmalloc/libpas/src/libpas/pas_probabilistic_guard_malloc_allocator.h

```diff
@@ -28,22 +28,48 @@
 #define PAS_PROBABILISTIC_GUARD_MALLOC_ALLOCATOR
 
 #include "pas_utils.h"
+#include "pas_large_heap.h"
+#include <stdbool.h>
 
 PAS_BEGIN_EXTERN_C;
 
-/* Initial Function Definitions */
+// structure for holding metadata of pgm allocations
+typedef struct pgm_storage pgm_storage;
+struct pgm_storage {
+    size_t allocation_size_requested;
+    size_t size_of_data_pages;
+    size_t mem_to_waste;
+    size_t mem_to_alloc;
+    uintptr_t start_of_data_pages;
+    uintptr_t upper_guard_page;
+    uintptr_t lower_guard_page;
```

Comment:
I think only necessary thing in this storage will be `allocation_size_requested`. The other values are only for debugging purpose or values that can be calculated from `pointer` & `allocation_size_requested`.
Because utility heap is precious resource in bmalloc (it is using limited compact heap), in the production version of PGM in libpas, we should not allocate it for each allocation. So, please keep in mind that we need to make this struct just the following in the release build.

```
struct pas_pgm_storage {
    size_t allocation_size_requested;
};
```

And then, we do not need to have storage allocation via utility heap since pas_ptr_hash_map can store 64bit value as a value.